iroh/src/endpoint.rs

```diff
@@ -308,7 +308,8 @@ impl Builder {
     /// the endpoint. Discovery services that need to publish information need
     /// to use this secret key to sign the information.
     ///
-    /// If you add multiple discovery services, they will be combined using a
+    /// The internals of the [`Builder`] use the discovery objects passed to this function and
+    /// automatically combines them upon [`Builder::bind`] via a
```

Comment:
> If you call this method multiple times, all discovery services passed in will be combined via an internal [`crate::discovery::ConcurrentDiscovery`].  To remove all discovery services already added to the builder use [`Self::clear_discovery`].

Would be my attempt.  And that would remove the separate paragraph about clearing below.

Would be great if you could also fix the wrong function name in the paragraph above and link it to [`IntoDiscovery::into_discovery`]